### ISSUE TITLE
Error with DROP PROCEDURE IF EXISTS

### ISSUE THREAD

**Issue by ypereirareis** [zendesk/maxwell#1393]
Hi,

Thank you again for this wonderful "database event stream tool" !!

I had a strange error a few days ago with a `DROP PROCEDURE IF EXISTS` statement.
* Running with docker : ```FROM zendesk/maxwell:v1.22.6```
* I had to skip this statement incrementing to the next binlog position.
* Sorry if it's already fixed in a more recent version.

Here are the error logs (sorry for the order, all lines timestamped at the same millisecond in our logging stack)

```
November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.Maxwell.start(Maxwell.java:156)
    November 7th 2019, 09:00:06.429 com.zendesk.maxwell.schema.ddl.MaxwellSQLSyntaxError: DROP
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:222)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.schema.ddl.SchemaChange.parse(SchemaChange.java:101)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.schema.AbstractSchemaStore.resolveSQL(AbstractSchemaStore.java:49)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.schema.MysqlSchemaStore.processSQL(MysqlSchemaStore.java:102)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:222) ~[maxwell-1.22.6.jar:1.22.6]
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.Maxwell.main(Maxwell.java:243) ~[maxwell-1.22.6.jar:1.22.6]
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.schema.ddl.MysqlParserListener.visitErrorNode(MysqlParserListener.java:93)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.Maxwell.main(Maxwell.java:243)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.Maxwell.start(Maxwell.java:156) ~[maxwell-1.22.6.jar:1.22.6]
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34) ~[maxwell-1.22.6.jar:1.22.6]
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.schema.ddl.SchemaChange.parseSQL(SchemaChange.java:89)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:177)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34)
    November 7th 2019, 09:00:06.429     at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
    November 7th 2019, 09:00:06.429     at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:313)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:335)
    November 7th 2019, 09:00:06.429     at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:607)
    November 7th 2019, 09:00:06.429     at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:41)
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:313)
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:335)
    November 7th 2019, 09:00:06.428 08:00:04,867 ERROR AbstractSchemaStore - Error on bin log position Position[BinlogPosition[mysql-bin.068121:5396538], lastHeartbeat=0]
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.Maxwell.start(Maxwell.java:156)
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.schema.ddl.SchemaChange.parseSQL(SchemaChange.java:89) ~[maxwell-1.22.6.jar:1.22.6]
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.schema.ddl.MysqlParserListener.visitErrorNode(MysqlParserListener.java:93) ~[maxwell-1.22.6.jar:1.22.6]
    November 7th 2019, 09:00:06.428 com.zendesk.maxwell.schema.ddl.MaxwellSQLSyntaxError: DROP
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.replication.BinlogConnectorReplicator.work(BinlogConnectorReplicator.java:177) ~[maxwell-1.22.6.jar:1.22.6]
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.schema.ddl.SchemaChange.parse(SchemaChange.java:101)
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.Maxwell.startInner(Maxwell.java:222)
    November 7th 2019, 09:00:06.428     at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52) ~[antlr4-runtime-4.5.jar:4.5]
    November 7th 2019, 09:00:06.428 com.zendesk.maxwell.schema.ddl.MaxwellSQLSyntaxError: DROP
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:34)
    November 7th 2019, 09:00:06.428     at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:41)
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.Maxwell.main(Maxwell.java:243)
    November 7th 2019, 09:00:06.428     at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52) ~[antlr4-runtime-4.5.jar:4.5]
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.replication.BinlogConnectorReplicator.processQueryEvent(BinlogConnectorReplicator.java:313) ~[maxwell-1.22.6.jar:1.22.6]
    November 7th 2019, 09:00:06.428 DROP PROCEDURE IF EXISTS `init_uuid_databases`'
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.replication.BinlogConnectorReplicator.getRow(BinlogConnectorReplicator.java:607)
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.schema.AbstractSchemaStore.resolveSQL(AbstractSchemaStore.java:49) ~[maxwell-1.22.6.jar:1.22.6]
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.schema.AbstractSchemaStore.resolveSQL(AbstractSchemaStore.java:49)
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.schema.MysqlSchemaStore.processSQL(MysqlSchemaStore.java:102)
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.schema.ddl.SchemaChange.parseSQL(SchemaChange.java:89)
    November 7th 2019, 09:00:06.428     at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:41) ~[antlr4-runtime-4.5.jar:4.5]
    November 7th 2019, 09:00:06.428     at com.zendesk.maxwell.schema.ddl.MysqlParserListener.visitErrorNode(MysqlParserListener.java:93)
```

Thanks !!

**Comment by mpeyper** [comment]
We've just been hit by this as well in `v1.20.0`.

**Comment by scooterGrun** [comment]
As @mpeyper mentioned we ran into this issue today, which we've found to be caused by having SQL comments before the drop procedure statement

```
# some more bizzare text;
 DROP PROCEDURE IF EXISTS  `some_proc`;
```

**Comment by ypereirareis** [comment]
Hello,
Ok, in my case (if it can help), the exact statements causing the problem seems to be (note the comment after the use) :

```sql
use `my_database`; # PROCEDURE
DROP PROCEDURE IF EXISTS `init_uuid_databases`;
```

Thanks and have a good day !

**Comment by osheroff** [comment]
fixed in 1.24.0
